crates/core/machine/src/syscall/precompiles/u256x2048_mul/air.rs

```diff
@@ -0,0 +1,404 @@
+                        }
+
+                        for i in 0..WORDS_FIELD_ELEMENT {
+                            cols.hi_memory[i]
+                                .populate(event.hi_memory_records[i], &mut new_byte_lookup_events);
+                        }
+
+                        let a = BigUint::from_bytes_le(&words_to_bytes_le::<32>(&event.a));
+                        let b_array: [BigUint; 8] = event
+                            .b
+                            .chunks(8)
+                            .map(|chunk| BigUint::from_bytes_le(&words_to_bytes_le::<32>(chunk)))
+                            .collect::<Vec<_>>()
+                            .try_into()
+                            .unwrap();
+
+                        let effective_modulus = BigUint::one() << 256;
+
+                        let mut carries = vec![BigUint::zero(); 9];
+                        let mut ab_plus_carry_cols = [
+                            &mut cols.a_mul_b1,
+                            &mut cols.ab2_plus_carry,
+                            &mut cols.ab3_plus_carry,
+                            &mut cols.ab4_plus_carry,
+                            &mut cols.ab5_plus_carry,
+                            &mut cols.ab6_plus_carry,
+                            &mut cols.ab7_plus_carry,
+                            &mut cols.ab8_plus_carry,
+                        ];
+
+                        for (i, col) in ab_plus_carry_cols.iter_mut().enumerate() {
+                            let (_, carry) = col.populate_mul_and_carry(
+                                &mut new_byte_lookup_events,
+                                event.shard,
+                                &a,
+                                &b_array[i],
+                                &carries[i],
+                                &effective_modulus,
+                            );
+                            carries[i + 1] = carry;
+                        }
+                        row
+                    })
+                    .collect::<Vec<_>>();
+                records.add_byte_lookup_events(new_byte_lookup_events);
+                (rows, records)
+            })
+            .collect::<Vec<_>>();
+
+        //  Generate the trace rows for each event.
+        let mut rows = Vec::new();
+        for (row, mut record) in rows_and_records {
+            rows.extend(row);
+            output.append(&mut record);
+        }
+
+        pad_rows_fixed(
+            &mut rows,
+            || {
+                let mut row: [F; NUM_COLS] = [F::zero(); NUM_COLS];
+                let cols: &mut U256x2048MulCols<F> = row.as_mut_slice().borrow_mut();
+
+                let x = BigUint::zero();
+                let y = BigUint::zero();
+                let z = BigUint::zero();
+                let modulus = BigUint::one() << 256;
+
+                cols.a_mul_b1.populate(&mut vec![], 0, &x, &y, FieldOperation::Mul);
```

Comment:
Why do you need to call these `populate`s in the padded rows?

Reply:
If not, it fails